### DISCUSSION TITLE
Dynamically add auth headers to rewrites

### DISCUSSION THREAD

**Discussion by stutrek** [vercel/next.js#28639] [Ideas]
### Describe the feature you'd like to request

My back end, like most back ends, uses an authorization header. It also generates images that should not be public. I would like to drop them in an `img` tag for rendering, but I can't because they require an auth header.

I have access to the auth token in the session, I would like to add it to the request on the server side, at the rewrite stage.

### Describe the solution you'd like

I would like a headers callback on rewrites that allows me to add the auth header to the request.

### Describe alternatives you've considered

I will probably make an api route that proxies requests.

**Comment by rachel-church** [comment]
Similar to https://github.com/vercel/next.js/discussions/19078
